### ISSUE TITLE
reaching 100% CityGML 3.0 conformance

### ISSUE THREAD

**Issue by DiegoVinasco** [VCityTeam/UD-Graph#37]
tl;dr: add `rule-owl-prop-iso191502AssociationName` and `rule-owl-prop-iso191502Aggregation` to the proposed Shapechange configuration files, update the proposed data models with the new config.

In order to be officially 100% conformant to the CityGML 3.0 standard ALL conformance requirements must be satisfied ([CityGML Implementation Specification conformance statement](https://docs.ogc.org/is/20-010/20-010.html#toc11)):

The [Abstract Test suite](https://docs.ogc.org/is/20-010/20-010.html#toc72) contains a list of all requirements for every module and potential ADE, including example conformance tests and test methods.

For example Abstract Test A.1 states:

Requirement | /req/core/classes
-- | --
Test purpose | To validate that the Implementation Specification correctly implements the UML Classes defined in the Conceptual Model.
Test method type | Manual Inspection
Test method | For each UML class defined or referenced in the Core Package:<ol><li>Validate that the Implementation Specification contains a data element which represents the same concept as that defined for the UML class.</li><li>Validate that the data element has the same relationships with other elements as those defined for the UML class. Validate that those relationships have the same source, target, direction, roles, and multiplicies as those documented in the Conceptual Model.</li><li>Validate that the data element has the same properties (attributes) as those specified for the UML class. Validate that those properties have the same name, definition, type, and multiplicity of those documented in the Conceptual Model.</li><li>Validate that the properties of the data element include those of all superclasses of the UML class as documented in the Conceptual Model. Validate that those properties have the same name, definition, type, and multiplicity of those documented in the Conceptual Model.</li><li>Validate that the associations represented for the data element include those of all superclasses of the UML class as documented in the Conceptual Model. Validate that those representations have the same source, target, roles, and multiplicity of those documented in the Conceptual Model.</li><li>Validate that the Implementation Specification enforces all constraints imposed on the UML class by the Conceptual Model.</li></ol>

Although the current proposed CityGML OWL model meets most of these test requirements, we are not currently transforming association **roles** to OWL. According to ISO19150-2, associations should be transformed into rdfs:Properties (which we currently do). However association **roles** - such as aggregations and compositions - do not exist in OWL. ISO 19150-2 suggests storing the semantics of these roles as annotations properties `iso19150-2:aggregationType` and `iso19150-2:associationName`.

These can be fixed by adding the shapechange rules `rule-owl-prop-iso191502AssociationName` and `rule-owl-prop-iso191502Aggregation` to the proposed configuration.

**Comment by nataschake** [comment]
Hi, @DiegoVinasco!
I use this issue due to its title. 
In [CityGML3.0-ConceptualModel](https://github.com/opengeospatial/CityGML-3.0CM/releases/download/3.0.0-final.2021.02.23/CityGML_3.0_UML-Diagrams.pdf) file the feature type `AbstractFeatureWithLifespan` has four datatype properties ranging in DateTime: creationDate, terminationDate,valiedFrom, validTo. In one of your scripts, [update-triples.sh](https://github.com/VCityTeam/UD-Reproducibility/blob/master/Computations/RDF/CityOWL/update-triples.sh) one of actions is to remove exactly the statements about it:
```
python update_graph.py ../CityOWL/core.ttl ../CityOWL/core.ttl \
   'PREFIX owl:  <http://www.w3.org/2002/07/owl#>
    PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
    PREFIX xsd:  <http://www.w3.org/2001/XMLSchema#>
    PREFIX core: <https://www.opengis.net/ont/citygml/core/>

    DELETE DATA {
        core:AbstractFeatureWithLifespan.validFrom a owl:DatatypeProperty ;
            rdfs:range xsd:dateTime .
        core:AbstractFeatureWithLifespan.validTo a owl:DatatypeProperty ;
            rdfs:range xsd:dateTime .
        core:AbstractFeatureWithLifespan.creationDate a owl:DatatypeProperty ;
            rdfs:range xsd:dateTime .
        core:AbstractFeatureWithLifespan.terminationDate a owl:DatatypeProperty ;
            rdfs:range xsd:dateTime .
    }'
```
At the same time in [alignments.ttl](https://github.com/VCityTeam/UD-Reproducibility/blob/master/Computations/RDF/CityOWL/additional-triples/alignments.ttl) file the definition is different
```
core:validFrom a owl:ObjectProperty ;
    rdfs:subPropertyOf time:hasBeginning ;
    rdfs:range time:Instant .
core:validTo a owl:ObjectProperty ;
    rdfs:subPropertyOf time:hasEnd ;
    rdfs:range time:Instant .
core:creationDate a owl:ObjectProperty ;
    rdfs:subPropertyOf time:hasBeginning ;
    rdfs:range time:Instant .
core:terminationDate a owl:ObjectProperty ;
    rdfs:subPropertyOf time:hasEnd ;
    rdfs:range time:Instant .
```
so opening `core.ttl` in Protege results in errors (punning):
`Illegal redeclarations of entities: reuse of entity https://www.opengis.net/ont/citygml/core/terminationDate in punning not allowed [Declaration(ObjectProperty(<https://www.opengis.net/ont/citygml/core/terminationDate>)), Declaration(DataProperty(<https://www.opengis.net/ont/citygml/core/terminationDate>))]`

Why do you decide to align these properties to OWL-Time? And don't you plan to return to initial encoding where the properties mentioned above are Datatype Properties?
I found also [https://github.com/VCityTeam/UD-Reproducibility/issues/43](https://github.com/VCityTeam/UD-Reproducibility/issues/43), so I guess the decision is already made.
